controllers/storagecluster/placement.go

```diff
@@ -22,6 +22,11 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 		(&in).DeepCopyInto(&placement)
 	}
 
+	// default value for mds pod anti affinity is set to the filesystem name
+	if component == "mds" && placement.PodAffinity != nil {
```

Comment:
What if `placement.PodAffinity` is nil, The value will always be `ocs-storagecluster-cephfilesystem` . Is that okay?

Reply:
I think how ocs operator appends it wont be ever null.
I have added this check because of the test cases we have in the test file.

Reply:
I will be nil. The default set `PodAntiAffinity` not `PodAffinity` so it will be nil if we are using defaults. Moreover, in the case of customer-specified tolerations, we never know.

Reply:
actually I need to check `PodAntiAffinity `

Reply:
Now the default case is solved. 

What about the customer given tolerations? Do we need a value if the tolerations are given by the customer and `PodAntiAffinity` is not set? What if customer sets more than one item in the list?

Reply:
I think it will work like the default case, and later if there is a user input we have a later checks for that in the same function,

I will look again into it if nothing breaks

Reply:
This function will take care of all the caseshttps://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/placement.go#L115
And we should always add the deafult values as per design, as in what ways we want to restrict,
If we wanna override this function should be changed,
Which been called after I have set the deafult values